### PR TITLE
Fix build dependencies and library usage

### LINKED ISSUE #13: Let users pass in smartcrop options
Like faceDetectionHaarCascade, useFaceDetect etc.

### PR DESCRIPTION
**Pull request by sharpner**
draw2d master is required since the latest stable v1
still uses googlecode.

I added the haarscade file to the repository since the copyright of intel allows redistribution. 
Otherwise smartcrop would only work on linux.

But it currently won't work outside of this repository, I'm still trying to find a generic solution.

fixes #13 we can add more of the const options in other PRs.

### CODE REVIEW

**Comment by sharpner** [comment]
If you don't want the haarcascade file in the repository I could extract it to be configurable, but as long as the path is a hardcoded constant it's impossible to use on mac os x :/

File: crop.go [hunk 1]
```diff
@@ -41,16 +41,17 @@ import (
...
 	"github.com/nfnt/resize"
 )
 
 var skinColor = [3]float64{0.78, 0.57, 0.44}
 
 const (
-	detailWeight = 0.2
-	faceDetectionHaarCascade = "/usr/share/opencv/haarcascades/haarcascade_frontalface_alt.xml"
+	detailWeight             = 0.2
+	faceDetectionHaarCascade = "./files/haarcascade_frontalface_alt.xml"
```
Comment (muesli) [inline crop.go]: I'm not sure it's the best idea to ship the opencv haarcascades directly in smartcrop. We probably want to let the smartcrop user pass in their own xml path.

Comment (sharpner) [inline crop.go]: You are right. How do you think about breaking the API and do something like this:

```go
NewCropWithHaarCascade(filename string) (Crop, error) {
....
}

type Crop interface {
   SmartCrop(img image.Image, width, height int) (Crop, error)
}

That way the user must provide correct arguments at bootstrap and we can validate that everything works at bootstrap.

Currently if that file is not found, my whole application stops with os.Exit(1) which is hardcoded :)

Comment (sharpner) [inline crop.go]: It doesn't even have to be a breaking change, the old functionality can still be available.

Comment (muesli) [inline crop.go]: Sounds good to me. We should probably also have a NewCropper method without the haar-cascades that sets useFaceDetection to false.

Comment (sharpner) [inline crop.go]: I'll get to it!

Comment (muesli) [inline crop.go]: JFI: I don't mind breaking the compatibility if it makes for a nicer API. We can have a new stable release (0.2) afterwards.

Comment (sharpner) [inline crop.go]: okay. good to know.

**Comment by sharpner** [comment]
RFC for the last commit. What do you think? I tried to find a different naming to remove the stuttering with the package name. So that you can call `smartcrop.Find....` what do you think?

**Comment by sharpner** [comment]
I think I am done for now, this PR grows too much otherwise :/ Let me know what you think! :-)

File: crop.go [hunk 2]
```diff
@@ -25,31 +25,30 @@
...
 )
 
 var skinColor = [3]float64{0.78, 0.57, 0.44}
 
 const (
-	detailWeight = 0.2
+	detailWeight             = 0.2
 	faceDetectionHaarCascade = "/usr/share/opencv/haarcascades/haarcascade_frontalface_alt.xml"
 	//skinBias          = 0.01
 	useFaceDetection  = true // if true, opencv face detection is used instead of skin detection.
```
Comment (muesli) [inline crop.go]: We should probably remove these settings as they're now part of CropSettings.

Comment (sharpner) [inline crop.go]: I kinda liked it that you could see all defaults at one place, but of course I can simply move those to the default initializer.

**Comment by muesli** [comment]
Looking good! I think it'd be nice if we provide an example of using smartcrop with custom analyzer settings, too.

**Comment by sharpner** [comment]
I'll update the readme and will add a few Tests

**Comment by muesli** [comment]
You rock :+1:

**Pull request merged**